qlib/workflow/record_temp.py

```diff
@@ -141,6 +141,12 @@ def generate(self, **kwargs):
                 del params["data_key"]
                 # The backend handler should be DataHandler
                 raw_label = DatasetH.prepare(**params)
+
+            if not isinstance(raw_label, pd.DataFrame):
+                index = raw_label.get_index()
+                raw_label = raw_label.data.loc[index]
+                raw_label = raw_label.iloc[:,-1:]
```

Comment:
We should make less assumption to make this module more general 
The assumption we've made here is the `col_set=="label"`